test/test_distributions.py

```diff
@@ -203,6 +203,44 @@ def test_normal_sample(self):
                                         scipy.stats.norm(loc=mean, scale=std),
                                         'Normal(mean={}, std={})'.format(mean, std))
 
+    def test_exponential(self):
+        rate = Variable(torch.randn(5, 5).abs(), requires_grad=True)
+        rate_1d = Variable(torch.randn(1).abs(), requires_grad=True)
+        self.assertEqual(Exponential(rate).sample().size(), (5, 5))
+        self.assertEqual(Exponential(rate).sample((7,)).size(), (7, 5, 5))
+        self.assertEqual(Exponential(rate_1d).sample((1,)).size(), (1, 1))
+        self.assertEqual(Exponential(rate_1d).sample().size(), (1,))
+        self.assertEqual(Exponential(0.2).sample((1,)).size(), (1,))
+        self.assertEqual(Exponential(50.0).sample((1,)).size(), (1,))
+
+        self._gradcheck_log_prob(Exponential, (rate,))
+        state = torch.get_rng_state()
+        eps = rate.new(rate.size()).exponential_()
+        torch.set_rng_state(state)
+        z = Exponential(rate).rsample()
+        z.backward(torch.ones_like(z))
+        self.assertEqual(rate.grad, -eps / rate**2)
+        rate.grad.zero_()
+        self.assertEqual(z.size(), (5, 5))
+
+        def ref_log_prob(idx, x, log_prob):
+            m = rate.data.view(-1)[idx]
+            expected = math.log(m) - m * x
+            self.assertAlmostEqual(log_prob, expected, places=3)
+
+        self._check_log_prob(Exponential(rate), ref_log_prob)
+
+    # This is a randomized test.
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_exponential_sample(self):
+        self._set_rng_seed()
+        for rate in [1e-5, 1.0, 10.]:
+            # NOTE: see note in distributions.exponential
+            self._check_sampler_sampler(Exponential(rate),
+                                        scipy.stats.expon(scale=1. / rate),
+                                        'Exponential(rate={})'.format(rate), failure_rate=5e-4)
```

Comment:
I think it's a safer policy to increment seeds of randomly failing tests (`self._set_rng_seed(1)` above), rather than increase `failure_rate`. It's nice to have a globally tunable failure rate, so that as a library and its tests scale by 10x and 100x, the failure rate can be correspondingly increased globally.